### DISCUSSION TITLE
Having issues adding VENV support

### DISCUSSION THREAD

**Discussion by creda-technologies** [zed-industries/zed#8266] [Feature Requests]
Hey all, I tried the solution in this discussion: https://github.com/zed-industries/community/issues/205, but am having issues getting it working with my project. I've tried both putting it into both the parent directory and the directory with all the code, and neither seem to work. Anyone know how to solve this?
<img width="1512" alt="Screenshot 2024-02-23 at 5 28 19 AM" src="https://github.com/zed-industries/zed/assets/160958497/b20c3931-d54c-45d7-b02a-fd4da4b7aedb">

**Comment by Moshyfawn** [comment]
I recently ran into a similar problem where Zed wouldn't recognise my .venv unless I explicitly specified the path to it. What worked for me was to define a `pyproject.toml` with this content:

```toml
[tool.pyright]
venvPath = "."
venv = ".venv"
```

**Reply by creda-technologies** [reply]
This worked! But now I have another similar problem. Pyright is unable to detect what the root folder is in my library giving me the following issue. Do you happen to have a solution for this? The code itself works correctly, but the import statements are showing up as incorrect in pyright. I tried to add the "root" attribute in [tool.pyright] to both "anchor_sdk" and ".", but was unsuccessful.

<img width="1624" alt="Screenshot 2024-02-27 at 4 23 16 AM" src="https://github.com/zed-industries/zed/assets/160958497/7b9e6a52-58d3-4b4a-b2fa-4a96cdc0ce1d">

**Comment by christiankriedemann** [comment]
If the `pyproject.toml` doesn't fix it, then add a `pyrightconfig.json` with this content:

```json
   {
     "venvPath": ".",
     "venv": ".venv"
   }
```

**Reply by ArunJRK** [reply]
I have both but still doesn't work. Tried both . and full path approaches

Just a thought could it be because we are in a different volume than default one on the disk

Edit: My bad was using the wrong interpreter

**Comment by sodimm** [comment]
I fixed this error by using the following in the settings file:
`  "lsp": {
    "pyright": {
      "settings": {
        "python": {
          "pythonPath": "venv/bin/python", # change your envelope path as required.
          "venvPath": "."
        }
      }
    }
  },
`

I didn't find this on my own, this [article](https://www.buzl.uk/2024/08/21/pyright.html) helped me finding a solution

**Reply by chandujr** [reply]
> I fixed this error by using the following in the settings file: ` "lsp": { "pyright": { "settings": { "python": { "pythonPath": "venv/bin/python", # change your envelope path as required. "venvPath": "." } } } },`
> 
> I didn't find this on my own, this [article](https://www.buzl.uk/2024/08/21/pyright.html) helped me finding a solution

This is not working for me. And I think the link to the article is dead now. By settings file, you meant the local project settings file, right? Not the zed config file?

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links
